### PR TITLE
removed lodash dependancy and updated to ES6

### PR DESCRIPTION
**Pull request by ja9adeesh**
removed `lodash` dependancy and updated to `ES6`, jus checking whether you are interested to merge or create a branch with this.

### CODE REVIEW

**Comment by bonzanni** [comment]
I would really like to see this merged.

**Comment by daviesgeek** [comment]
I really would too, but I don't have the time to review 😭

**Comment by bostrom** [comment]
Very nice! I'll try to have a look at it! The only thing I'm concerned about is the number of old pull requests that would need to be updated to ES6 in order to be merged. Perhaps that's only a good thing, tough. Thoughts?

File: .jshintrc [hunk 1]
```diff
@@ -17,7 +17,6 @@
   "browser": true,
   "globals": {
     "module": true,
-    "angular": true,
-    "_": true
+    "angular": true
```
Comment (ja9adeesh) [inline .jshintrc]: added `"esversion": 6` and `"globals": [ "define": true, "require": true]`. its lint free now

**Comment by bostrom** [comment]
I'm more inclined to keeping the ES6 version in a separate branch for now. @ja9adeesh are you willing to try to keep that branch in sync with the master branch?

**Comment by bostrom** [comment]
Created the [es6](https://github.com/mgonto/restangular/tree/es6) branch based on this PR for maintaining the ES6 version.

**Comment by Rafi993** [comment]
@bostrom  it would be awesome if a separate npm package is created for ES6 so that users who want can use it directly.

**Pull request closed without merging**